app/models/watched_encode.rb

```diff
@@ -32,23 +32,37 @@ class WatchedEncode < ActiveEncode::Base
     if Settings.encoding.derivative_bucket &&
        (Settings.encoding.engine_adapter.to_sym == :ffmpeg || is_a?(PassThroughEncode))
       bucket = Aws::S3::Bucket.new(name: Settings.encoding.derivative_bucket)
-      encode.output.collect! do |output|
-        file = FileLocator.new output.url
-        key = file.location.sub(/\/(.*?)\//, "")
-        obj = bucket.object key
 
-        if File.exist? file.location
-          obj.upload_file file.location
-          File.delete file.location
+      encode.output.collect! do |output|
+        if output.url.include?('supplemental_files')
+          new_file = SupplementalFile.new(tags: ['caption'], parent_id: record.master_file_id)
+          new_file.attach_file(FileLocator.new(output.url).location, io: true)
+          new_file.save
+          output.url = if Settings.active_storage.bucket.present?
+                      "s3://#{Settings.active_storage.bucket}/#{new_file.file.blob.key}"
+                    else
+                      new_file.file.blob.url
+                    end
+          output.id = new_file.id
```

Comment:
How about making this the global id?  Then it can be verified that the output is actually a SupplementalFile when adding it to the MasterFile.
```suggestion
          output.id = new_file.to_global_id.to_s
```